src/Fantomas.Core/CodePrinter.fs

```diff
@@ -1244,16 +1244,17 @@ and genExpr astContext synExpr ctx =
                 !- "for "
                 +> genPat astContext p
                 +> !- " in "
-                +> autoIndentAndNlnIfExpressionExceedsPageWidth (genExpr astContext e1)
-                +> ifElse
-                    isArrow
-                    (sepArrow
-                     +> autoIndentAndNlnIfExpressionExceedsPageWidth (genExpr astContext e2))
-                    (!- " do"
-                     +> indent
-                     +> sepNln
-                     +> genExpr astContext e2
-                     +> unindent)
+                +> leadingExpressionIsMultiline
+                    (autoIndentAndNlnIfExpressionExceedsPageWidth (genExpr astContext e1))
+                    (fun isMultiline ->
+                        (ifElse
```

Comment:
Avoid the nested `ifElse`, if you are already in a lambda writing `if ... then ... else` is a bit more readable.